packages/server/src/validators/UsersValidator.js

```diff
@@ -0,0 +1,27 @@
+import Models from '../models'
+import BaseValidator from './BaseValidator'
+import { isAddress, toChecksumAddress } from 'web3-utils'
+
+const { UserAddress } = Models
+
+class UsersValidator extends BaseValidator {
+  async validateForCreate({ email, address }) {
+    this._validateEmail(email)
+    await this._validateAddress(address)
+    return this.resetErrors()
+  }
+
+  _validateEmail(email) {
+    if (!email) this.addError({ email: 'An email address must be given' })
+  }
+
+  async _validateAddress(address) {
+    if (!isAddress(address)) return this.addError({ address: 'Given address is not valid' })
+
+    const checksumAddress = toChecksumAddress(address)
```

Comment:
What you think about storing addresses in the db as just lowercased addresses?

I've been thinking about this for a bit and more or less decided that it would be better if we:

- Sanity checked all user-generated addresses with checksums
- Only used lowercase addresses internally, once those checks have passed

Forgetting to do the `addressEquals()` check instead of a `===` check has bit people way too many times, and it's because we don't have strong responsibility rules around when we preserve and when we strip the checksum.

Reply:
Sounds great, let's do that!